components/governance/org.wso2.carbon.governance.lcm/src/main/java/org/wso2/carbon/governance/lcm/tasks/LCNotificationScheduler.java

```diff
@@ -0,0 +1,234 @@
+
+    private static final Log log = LogFactory.getLog(LCNotificationScheduler.class);
+
+    // Duration of a day 1 * 24 * 60 * 60 * 1000.
+    private final long durationOfADay = 86400000;
+
+    // Used to as Qname for checkpoint id.
+    private final String checkpointId = "id";
+
+    // Date separator
+    private final String dateSeparator = "-";
+
+    // Date length
+    private final int dateLength = 10;
+
+    /**
+     * This method is used to send notifications for checkpoints in lifecycle. This method is called using the
+     * scheduler task CheckpointNotificationSchedulerTask.
+     */
+    public void run() {
+        LifecycleNotificationDAO lifecycleNotificationDAO = new JDBCLifecycleNotificationDAOImpl();
+        ArrayList<LCNotification> notifications = new ArrayList<LCNotification>();
+
+        try {
+            notifications = lifecycleNotificationDAO
+                    .getValidNotifications(CommonUtil.getRegistryService().getRegistry());
+        } catch (GovernanceException e) {
+            log.error("Error while getting notifications. Hold notification sending job.", e);
+            /*
+             Throwing the exception is no need since this is a scheduler task and scheduler tasks uses
+             org.wso2carbon.ntask.core.Task interface which doesn't throw exceptions from execute() method signature.
+             */
+            return;
+        } catch (RegistryException e) {
+            log.error("Error while getting notifications to send Lifecycle checkpoint notifications.", e);
+        }
+
+        // Send notifications if exists.
+        if (notifications != null && notifications.size() > 0) {
+            // Parse this for a another method.
+            for (LCNotification schedulerBean : notifications) {
+                // Creating a Checkpoint notification event.
+                String notificationMessage = getNotificationMessage(schedulerBean);
+                LifecycleNotificationEvent<String> notificationEvent = new LifecycleNotificationEvent<String>
+                        (notificationMessage);
+                notificationEvent.setTenantId(schedulerBean.getTenantId());
+                notificationEvent.setResourcePath(schedulerBean.getRegPath());
+                // Sending notification using notification service.
+                NotificationService notificationService = new EventingServiceImpl();
+                try {
+                    notificationService.notify(notificationEvent);
+                    if (log.isDebugEnabled()) {
+                        log.debug("Notification '" + notificationMessage + "' sent to notification service.");
+                    }
+                    // Exception is caught because notificationService.notify method throws the exception from Exception
+                    // class. This will be refactored after fixing this JIRA: https://wso2.org/jira/browse/REGISTRY-2433
+                } catch (Exception e) {
```

Comment:
Thanks for the comment.
Exception is caught here because notificationService.notify method throws the exception from Exception class. This will be refactored after closing the public JIRA[1] after merging its pull request.
[1] https://wso2.org/jira/browse/REGISTRY-2433.